src/firefly/js/ui/FileUploadProcessor.jsx

```diff
@@ -27,26 +27,64 @@ const SUPPORTED_TYPES=[
     FileAnalysisType.REGION,
     FileAnalysisType.Image,
     FileAnalysisType.Table,
-    FileAnalysisType.Spectrum,
-    FileAnalysisType.REGION,
+    FileAnalysisType.Spectrum
 ];
 
 export function resultSuccess(request) {
 
-    const currentReport = request.additionalParams?.currentReport;
-    const currentDetailsModel = request.additionalParams?.currentDetailsModel;
-    const summaryModel = request.additionalParams?.summaryModel;
-    const groupKey = request.additionalParams?.groupKey;
-    const acceptMoc = request.additionalParams?.acceptMoc;
+    const {currentReport, currentDetailsModel, summaryModel, groupKey, acceptList, uniqueTypes}= request.additionalParams ?? {};
     const summaryTblId = groupKey; //FileUploadAnalysis
 
-    const isTablesOnly= () => getAppOptions()?.uploadPanelLimit==='tablesOnly';
-    if (isTablesOnly()) return tablesOnlyResultSuccess(request, summaryTblId, currentReport, summaryModel, groupKey);
+    if (acceptOnlyTables(acceptList)) return tablesOnlyResultSuccess(request, summaryTblId, currentReport, summaryModel, groupKey);
     const fileCacheKey = getFileCacheKey(groupKey);
 
     const tableIndices = getSelectedRows(FileAnalysisType.Table, summaryTblId, currentReport, summaryModel);
     const imageIndices = getSelectedRows(FileAnalysisType.Image, summaryTblId, currentReport, summaryModel);
 
+    if (isRegion(summaryModel) && !acceptRegions(acceptList)) {
+        showInfoPopup('You may not load a Region file from here.', 'File Type Mismatch');
+        return false;
+    }
+
+    //uniqueTypes are the types of the parts of the uploaded file (retrieved in FileAnalysis)
+    if (uniqueTypes.includes(IMAGES) && uniqueTypes.includes(TABLES)) {
```

Comment:
I got stuck here.  The logic is very complicated.  In this block, `uniqueTypes` is only checked against `IMAGES` and `TABLES`.  But elsewhere, it may contains other values, like `uniqueTypes.includes(REGIONS)`.  Also, is `uniqueTypes` good for validation?  
I have a hard time distinguishing validation logic versus processing logic.

For validation, I would expect to see `acceptList` match against the type returned from each selected row.
Of course, if the file is a primary data and not a table of data products, then it should validate against that.

The type returned from each row should be the most detailed type, i.e.  Spectrum, Region, etc.  
The type in `acceptList` can be a superclass, like Tables.

Once pass validation, then I would expect to see a loop to fire off action(s) based on the selected type.
I believe if done right, it would simplify the code, at least for me.